ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -511,12 +507,28 @@ private boolean isSourceTableMaterializationEnabled() {
         .getBoolean(KsqlConfig.KSQL_SOURCE_TABLE_MATERIALIZATION_ENABLED);
   }
 
+  private SourceName getSourceName(final ConfiguredStatement<?> statement) {
```

Comment:
return `Optional<SourceName>` here instead of using `null` to signal no source name